geonode/commands/views.py

```diff
@@ -0,0 +1,86 @@
+#########################################################################
+#
+# Copyright (C) 2021 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+from rest_framework import status, views
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+
+from .tasks import enqueue_jobs
+from geonode.permissions import IsSuperUser
+
+class CommandView(views.APIView):
+    permission_classes = [IsAuthenticated, IsSuperUser]
+
+    def get(self, request):
+        # TODO: This may return a list of user's jobs
+        return Response(
+                {"message": "yet to be implemented!"},
+                status=status.HTTP_200_OK,
+            )
+
+    def post(self, request):
+        """
+        API to enque management commands into Celery as cron jobs
+
+        Expected Payload Format
+        {
+            "cmd": command name
+            "args": [<arg1>, <arg2>],
+            "kwargs: {<key1>: <val1>, <key2>: <val2>}
+        }
+
+        Example Payload:
+        {
+        "cmd": "test_command",
+        "args": ["delta"],
+        "kwargs": {
+                "cpair": [10, 40],
+                "ppair": ["a", "b"]
+            }
+        }
+        """
+        
+        args = request.data.get("args", [])
```

Comment:
We can think about having this part into a `serializer` with his own `validation` method

Reply:
But we are not using any model. That's why we are settled to `APIView` also.
Even otherwise, the `BaseCommand` class does the error handling, and that validation has specialization of such Python CLI args.

Reply:
Can't we use an abstract model to handle this part? Is just a matter of making it a bit more robust :)

Reply:
Again why we need to make abstract model when handling is already handled by `BaseCommand`.
And if it such necessary, best option is the model itself, and you have serializer also in the package.